pkg/controllers/dnset/configmap.go

```diff
@@ -0,0 +1,127 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dnset
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/matrixorigin/matrixone-operator/api/core/v1alpha1"
+	"github.com/matrixorigin/matrixone-operator/pkg/controllers/common"
+	"github.com/matrixorigin/matrixone-operator/pkg/utils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"text/template"
+)
+
+var startScriptTpl = template.Must(template.New("dnservice-start-script").Parse(`
+#!/bin/sh
+set -eu
+
+POD_NAME=${POD_NAME:-$HOSTNAME}
+ADDR="${POD_NAME}.${HEADLESS_SERVICE_NAME}.${NAMESPACE}.svc"
+ORDINAL=${POD_NAME##*-}
+UUID=$(printf '00000000-0000-0000-0000-%012x' ${ORDINAL})
+conf=$(mktemp)
+
+bc=$(mktemp)
+cat <<EOF > ${bc}
+uuid = "${UUID}"
+service-address = "0.0.0.0:{{ .DNServicePort }}"
+listen-address = "${ADDR}:{{ .DNServicePort }}"
```

Comment:
looks like we should swap the values of `service-address` and `listen-address` ?